app/components/TrustedSourceList.js

```diff
@@ -0,0 +1,88 @@
+import findIndex from 'lodash/findIndex';
+import React from 'react';
+import { StyleSheet, Switch, TouchableOpacity } from 'react-native';
+import { MenuOption } from 'react-native-popup-menu';
+
+import { Typography } from '../components/Typography';
+import Colors from '../constants/colors';
+import languages from '../locales/languages';
+
+export const TrustedSourceList = ({
+  selectedAuthorities,
+  authoritiesList,
+  isAuthorityFilterActive,
+  toggleFilterAuthoritesByGPSHistory,
+  addAuthorityToState,
+  filterAuthoritesByGPSHistory,
+  onMenuClick,
+}) => {
+  const isAuthoritySelected = name => {
+    return findIndex(selectedAuthorities, ['key', name]) < 0;
+  };
+
+  return (
+    <>
+      {__DEV__ && (
+        <TouchableOpacity
+          style={styles.authorityFilter}
+          onPress={() => toggleFilterAuthoritesByGPSHistory()}>
+          <Typography style={styles.authorityFilterText} use={'body2'}>
+            {languages.t('label.filter_authorities_by_gps_history')}
+          </Typography>
+          <Switch
+            onValueChange={val => filterAuthoritesByGPSHistory({ val })}
+            value={isAuthorityFilterActive}
+          />
+        </TouchableOpacity>
+      )}
+      {authoritiesList === undefined
+        ? null
+        : authoritiesList.map(item => {
```

Comment:
There's a lot goin on in this block. Hard to follow. 

This is `.filter()` operation:

```js
const availableAuthorities = authoritiesList.filter(item => !isAuthoritySelected(item));
```

do that above, and then it makes this much clearer:

```js
{availableAuthorities && availableAuthorities.map(item => <MenuOption ... />)}
```